File: Commander/Request/Request.cs
namespace Commander
{
    class Request
    {
        public Request(RequestFactory requestFactory, int number)
        {
            this.requestFactory = requestFactory;
            this.number = number;
        }

        public bool IsCancelled => number < requestFactory.Number;

        readonly RequestFactory requestFactory;
        readonly int number;
    }
}

File: Commander/Request.fs
namespace Commander 

open System.Threading

type Request(requestFactory: RequestFactory, number) = 
    member this.IsCancelled = number < requestFactory.Number

and RequestFactory() = 
    let mutable number = 0

    member this.Number with get () = number

    member this.create () = 
        Request(this, Interlocked.Increment(ref number)) 

